stixcore/processing/level0.py

```diff
@@ -0,0 +1,365 @@
+        filtered_packets[value] = list(filter(
+            lambda x: x.spid == value and isinstance(x, GenericTMPacket), packet_list))
+
+    complete = defaultdict(list)
+    incomplete = defaultdict(list)
+
+    for key, packets in filtered_packets.items():
+        sequences = extract_sequences(packets[:])
+        for seq in sequences:
+            if len(seq) == 1 and \
+                    seq[0].source_packet_header.sequence_flag == SequenceFlag.STANDALONE:
+                complete[key].append(seq)
+            elif (seq[0].source_packet_header.sequence_flag == SequenceFlag.FIRST
+                  and seq[0].source_packet_header.sequence_flag == SequenceFlag.LAST):
+                complete[key].append(seq)
+            else:
+                incomplete[key].append(seq)
+                logger.warning('Incomplete sequence for %d', key)
+
+    return complete, incomplete
+
+
+def extract_sequences(packets):
+    """
+    Extract complete and incomplete sequences of packets.
+
+    Packets can be either stand-alone, first ,continuation or last when TM is downloaded maybe
+    missing some packets so we we try to extract complete and incomplete sequences.
+
+    Parameters
+    ----------
+    packets : list
+        List of packets
+
+    Returns
+    -------
+    list
+        A list of packet packets
+    """
+    out = []
+    i = 0
+    cur_seq = None
+    while i < len(packets):
+        cur_packet = packets.pop(i)
+        cur_flag = cur_packet.source_packet_header.sequence_flag
+        if cur_flag == SequenceFlag.STANDALONE:
+            out.append([cur_packet])
+        elif cur_flag == SequenceFlag.FIRST:
+            cur_seq = [cur_packet]
+        if cur_flag == SequenceFlag.MIDDLE:
+            if cur_seq:
+                cur_seq.append(cur_packet)
+            else:
+                cur_seq = [cur_packet]
+        if cur_flag == SequenceFlag.LAST:
+            if cur_seq:
+                cur_seq.append(cur_packet)
+                out.append(cur_seq)
+            else:
+                out.append([cur_packet])
+            cur_seq = None
+
+    if cur_seq:
+        out.append(cur_seq)
+
+    return out
+
+
+def process_packets(packet_lists, spid, product, basepath=None, status='', overwrite=False,
+                    use_name=False):
+    """
+    Process a sequence containing one or more packets for a given product.
+
+    Parameters
+    ----------
+    packet_lists : list
+        List of packet sequences
+    spid : int
+        SPID
+    product : basestring
+        Product name
+    basepath : pathlib.Path
+        Path
+    overwrite : bool (optional)
+        False (default) will raise error if fits file exits, True overwrite existing file
+    """
+    fits_processor = FitsL1Processor(basepath)
+
+    # For HK merge all stand alone packets in request
+    if spid in [54101, 54102]:
+        packet_lists = [list(chain.from_iterable(packet_lists))]
+
+    prod = None
+
+    for packets in packet_lists:
```

Comment:
This could be a factory like the Packet get rid of all this if else and allow up to add more without changing code